src/backend/env/memory.rs

```diff
@@ -362,18 +362,13 @@ impl<K: Ord + Eq, T: Serialize + DeserializeOwned> Default for ObjectManager<K,
     }
 }
 
-impl<K: Eq + Ord + Clone + Display, T: Serialize + DeserializeOwned> ObjectManager<K, T> {
+impl<K: Eq + Ord + Clone + Display, T: Default + Serialize + DeserializeOwned> ObjectManager<K, T> {
     pub fn len(&self) -> usize {
         self.index.len()
     }
 
     pub fn insert(&mut self, id: K, value: T) -> Result<(), String> {
-        assert!(self.initialized, "allocator uninitialized");
-        if self.index.contains_key(&id) {
-            self.remove(&id)?;
-        }
-        self.index.insert(id, self.api.borrow_mut().write(&value)?);
-        Ok(())
+        unimplemented!()
```

Comment:
Are you sure this is not going to trigger a panic in the pre-upgrade? Or prevent emergency upgrades somehow? I cannot foresee the consequences and side-effects of this change, so it seems dangerous to me.

Reply:
It should not, but at the same time we can simply do nothing, not panic. Will change!